cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -17,53 +15,234 @@
+                comment = args[++i];
+            } else if (args[i].equals("--createNewSBOM")) {
+                cmd = "createNewSBOM";
+            } else if (args[i].equals("--addMetadata")) {        //This is Metadata Component. We can set "name" for Metadata->Component.
+                cmd = "addMetadata";
+            } else if (args[i].equals("--addMetadataProp")) {   //This is MetaData Component --> Property -> name-value: os, arch, variant, scmRef, buildRef, full_version_output, makejdk_any_platform_args
+                cmd = "addMetadataProperty";
+            } else if (args[i].equals("--addComponent")) {      //This is Components->Property: will add name-value.
+                cmd = "addComponent";
+            } else if (args[i].equals("--addComponentProp")) {  //This is Components->Property: will add name-value.
+                cmd = "addComponentProp";
+            } else if (args[i].equals("--addExternalRef")) {
+                cmd = "addExtRef";
+            } else if (args[i].equals("--addComExtRef")) {
+                cmd = "addComExtRef";
+            }
         }
-        System.out.println("");
+        switch (cmd) {
+            case "createNewSBOM": {                           //Creates JSON file
+                Bom bom = createBom(name, version);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
+
+            case "addMetadata": {                             //This is Metadata Component --> name
+                Bom bom = readJSONfile();
+                bom = addMetadata(bom, name);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
+
+            case "addMetadataProperty": {                     //This is MetaData--> Component --> Property -> name-value:
+                Bom bom = readJSONfile();
+                bom = addMetadataProperty(bom, name, value);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
 
-        Bom bom = createTestBom();
-        String json = generateBomJson(bom);
+            case "addComponent": {                              //This is Components adds name-value pairs to List
+                Bom bom = readJSONfile();
+                bom = addComponent(bom, name);
+                String json = generateBomJson(bom);
+                writeJSONfile(json, fileName);
+                System.out.println("SBOM: " + json);
+            } break;
 
-        System.out.println("SBOM: " + json);
+            case "addComponentProp": {                             //This is Components adds name-value pairs to List
+                Bom bom = readJSONfile();
+                bom = addComponentProperty(bom, name, value);
```

Comment:
needs a "compRef" param for which Component to add prop to..

Reply:
@andrew-m-leonard I will work on this again to see if we can add multiple Components.

Reply:
@SehrishHussain don't worry about multiple Components in this PR